### ISSUE TITLE
PlayerView: Provide option to show default artwork when no prepared player is attached

### ISSUE THREAD

**Issue by roninshahi** [google/ExoPlayer#6265]
hi
I wanna show a preview bitmap of a video in exoplayer.i have the bitmap and I use 
```
playerview.setUseArtwork(true);
playerview.setDefaultArtwork(bitmap); 
```
but it doesn't work.
***i havent write any code to connect playerview to exoplayer.
I used an image view to show the preview and then hide it but it wasn't good.

**Comment by ojw28** [comment]
Artwork passed to `setDefaultArtwork` is only displayed when the player is connected to the view and prepared (and if the prepared media doesn't contain any artwork of its own).

I'll mark this as an enhancement to provide an option for the default artwork to also be displayed when there's no player and when the player is not prepared.

**Comment by roninshahi** [comment]
ty

**Comment by HiroyukiTamura** [comment]
> I'll mark this as an enhancement to provide an option for the default artwork to also be displayed when there's no player and when the player is not prepared.

any updates?

**Comment by KgotsoK** [comment]
This should be prioritized...

**Comment by elmateo487** [comment]
This seems like such low hanging fruit. In order to show an image on the view, such as when there is no internet and the stream isn't loading, we have to wrap the view with another image view. IMO, this view should just be displayed the second the player is displayed. Then we can listen for playback errors and show the whole PlayerView, which would then have an image on it

**Comment by grndvl1** [comment]
Yeah, I'm surprised there is no way to show a placeholder image then when prepare and play go, hide that image and show video.  Should be an option unless there is something blocking that from happening.  I will just wrap an image view and show hide when player is prepping.

**Comment by lmj0011** [comment]
https://github.com/google/ExoPlayer/issues/4639

**Comment by ParticleCore** [comment]
this issue was opened in 2019. 
it's currently 2021, almost ending, and still no proper solution for this. forcing the use of an artwork by means of loading a chunk of video data can be quite expensive if it is a high resolution video, instead of just keeping the player unprepared and showing a custom image.

**Comment by grndvl1** [comment]
> this issue was opened in 2019. it's currently 2021, almost ending, and still no proper solution for this. forcing the use of an artwork by means of loading a chunk of video data can be quite expensive if it is a high resolution video, instead of just keeping the player unprepared and showing a custom image.

Its google they work at their own pace.  Hence why I shy away from them when using new libs.  Have you tried changing the Closed Captions on a video?  Wow what an experience and a piece of bad coding searching for ways to do it and depending on versions of the library, functions may or may not be there.  It was a nightmare.  Anyway, I hope they prioritize this soon as its really something that should have been there from the start.

**Comment by ridanasif** [comment]
Any updates on this issue?

**Comment by SaNu-hIT** [comment]
Any update on this issue ?

**Comment by mohodroid** [comment]
Any update on this issue ?

**Comment by imansdn** [comment]
Aw, shoot it's still open, now I have to go for a workaround! =)

**Comment by nikmax42** [comment]
?

**Comment by Abu-Abdullah** [comment]
is this repository tracked or we need to open this case again in the other active one:
https://github.com/androidx/media/issues

**Comment by icbaker** [comment]
@Abu-Abdullah Existing issues on this tracker are still monitored (hi!). If you would like to, you are welcome to file this issue again on https://github.com/androidx/media/issues and we will close this one as a duplicate of the new one. We have no current plans to implement this, and we are in the process of building a [suite of Compose UI components](https://github.com/androidx/media/tree/release/libraries/ui_compose) - so we are not actively improving `PlayerView` at the moment (and filing a new issue is unlikely to change our prioritization here).
